Source/DafnyLanguageServer.Test/Util/TestNotificationReceiver.cs

```diff
@@ -26,46 +25,43 @@ public class TestNotificationReceiver<TNotification> {
     }
 
     public void NotificationReceived(TNotification request) {
-      logger.LogTrace($"Received {request.Stringify()}");
       Assert.NotNull(request);
-      notifications.Enqueue(request);
-      notificationHistory.Add(request);
-      availableNotifications.Release();
+      logger.LogTrace($"Received {request.Stringify()}");
+
+      lock (this) {
+        notificationHistory.Add(request);
+        notifications.Enqueue(request);
```

Comment:
With the lock, I see how if notifications is a suffix of history, and then the postcondition keeps that property. We did not have this before the lock.

Could you have written it like this:
```suggestion
        History.Add(request);
        notifications.Enqueue(request);
```

so that it's consistent with `GetLatestAndClearQueue`?

Reply:
`History` is read only, so I can not call `Add` on it.